templates/asset_form.html.twig

```diff
@@ -367,6 +598,141 @@
+
+               {% if item.isField('periodicity') %}
+                  {{ fields.dropdownNumberField('periodicity', item.fields['periodicity'], __('Contract renewal period'), {
+                     'rand': rand,
+                     'min': 12,
+                     'max': 60,
+                     'step': 12,
+                     'toadd': {
+                        0: constant('Dropdown::EMPTY_VALUE'),
+                        1: _n('%d month', '%d months', 1)|format(1),
+                        2: _n('%d month', '%d months', 2)|format(2),
+                        3: _n('%d month', '%d months', 3)|format(3),
+                        6: _n('%d month', '%d months', 6)|format(6),
+                     },
+                     'unit': 'month',
+                  }) }}
+               {% endif %}
+
+               {% if item.isField('billing') %}
+                  {{ fields.dropdownNumberField('billing', item.fields['billing'], __('Invoice period'), {
+                     'rand': rand,
+                     'min': 12,
+                     'max': 60,
+                     'step': 12,
+                     'toadd': {
+                        0: constant('Dropdown::EMPTY_VALUE'),
+                        1: _n('%d month', '%d months', 1)|format(1),
+                        2: _n('%d month', '%d months', 2)|format(2),
+                        3: _n('%d month', '%d months', 3)|format(3),
+                        6: _n('%d month', '%d months', 6)|format(6),
+                     },
+                     'unit': 'month',
+                  }) }}
+               {% endif %}
+
+               {% if item.isField('renewal') %}
+                  {{ fields.dropdownArrayField('renewal', item.fields['renewal'], {
+                     (constant('Contract::RENEWAL_NEVER')): __('Never'),
+                     (constant('Contract::RENEWAL_TACIT')): __('Tacit'),
+                     (constant('Contract::RENEWAL_EXPRESS')): __('Express'),
+                  }, __('Renewal')) }}
+               {% endif %}
+
+               {% if item.isField('max_links_allowed') %}
+                  {{ fields.numberField('max_links_allowed', item.fields['max_links_allowed'], __('Max number of items'), {
+                     'min': 1,
+                     'max': 200000,
+                     'step': 1,
+                     'toadd': {0: __('Unlimited')}
+                  }) }}
+               {% endif %}
+
+               {% if item.isField('alert') and Entity_getUsedConfig('use_contracts_alert') %}
+                  {% set alert_html %}
+                     {{ displayLastAlert('Contract', item.fields['id']) }}
+                  {% endset %}
+                  {{ fields.dropdownArrayField('alert', item.fields['alert'], item.getAlertName(), __('Email alarms'), {
+                     'add_field_html': "<span class='text-muted'>" ~ alert_html ~ "</span>"
+                  }) }}
+               {% endif %}
+
+               {% if item.getType == 'Contract' %}
```

Comment:
regarding the whole part in the if block.
I rather move it into in a separate template included at the bottom of the page (before `components/form/buttons.html.twig` include)

and also insert the html into a div with `mx-n2` class to stick border of the tab

Reply:
also the display is alterered compared to previous html.
I think we need to have one colum with the yes/no part and another with the hours fields